capsules/src/rf233.rs

```diff
@@ -0,0 +1,882 @@
+#![allow(unused_parens)] // I like them sometimes, for formatting -pal
+
+///
+/// Capsule for sending 802.15.4 packets with an Atmel RF233.
+///
+/// This implementation is completely non-blocking. This means that
+/// the state machine is somewhat complex, as it must interleave interrupt
+/// handling with requests and radio state management. See the SPI
+/// read_write_done handler for details.
+///
+/// To do items:
+///    - Support TX power control
+///    - Support channel selection
+///    - Support link-layer acknowledgements
+///    - Support power management (turning radio off)
+// Capsule for sending 802.15.4 packets with an Atmel RF233.
+//
+// Author: Philip Levis
+// Date: Jan 12 2017
+//
+
+use core::cell::Cell;
+use kernel::common::take_cell::TakeCell;
+use kernel::hil::gpio;
+use kernel::hil::radio;
+use kernel::hil::spi;
+use kernel::returncode::ReturnCode;
+use rf233_const::*;
+
+const INTERRUPT_ID: usize = 0x2154;
+
+#[allow(unused_variables, dead_code,non_camel_case_types)]
```

Comment:
I want to keep UNKNOWN in so there is a pre-established way to handle cases when the radio goes into a problematic state. I agree that unused_variables should go away.